### PR TITLE
PP-747: Equivalence class tests with scheduler partitions

### PR DESCRIPTION
**Pull request by visheshh**
#### Bug/feature Description

[PP-747](https://pbspro.atlassian.net/browse/PP-747)

#### Affected Platform(s)
All platforms

#### Solution Description
Added additional tests for equivalence class with scheduler partitions

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by visheshh** [comment]
[multi_sched_equiv_class.txt](https://github.com/PBSPro/pbspro/files/1767550/multi_sched_equiv_class.txt)
Added test logs

File: test/tests/functional/pbs_multi_sched.py [hunk 1]
```diff
@@ -699,3 +707,415 @@ def test_cmp_fairshare(self):
...
+
+    def test_equiv_suspend_jobs(self):
+        """
+        Test that jobs fall into different equivalence classes
+        after they get suspended
+        """
+        self.setup_sc1()
+        self.setup_queues_nodes()
+        t = int(time.time())
+        self.scheds['sc1'].set_sched_config({'log_filter': 2048})
```
Comment (bhroam) [inline test/tests/functional/pbs_multi_sched.py]: This is done in setup_sc1() now.

**Comment by bhroam** [comment]
I suggest adding one more test.  Create two queues in the same partition with identical jobs submitted to each and see them fall into the same partition.

**Comment by bhroam** [comment]
Codacy is complaining about defaulting an argument to a dictionary.  This is because dictionaries are mutable times.  Every call to the submit_jobs() functions with the default attrs will point to the same memory.  If you ever change it, those changes will persist across calls.  Since attrs is read only, you should be fine.  If you want to quiet codacy, say attrs=None and then in the code say if attrs is None: attrs = {...}

**Comment by visheshh** [comment]
Hi @bhroam ,
I have addressed all the comments , Please review.
Thanks,
VIshesh

**Comment by visheshh** [comment]
squashed and rebased the commits , Please take this PR forward.

**Comment by bhroam** [comment]
@visheshh Travis failed due to your commit not being signed.

**Comment by mike0042** [comment]
TravisCI has failed. Please address the build issues.

**Comment by visheshh** [comment]
rebased and resolved conflicts

**Pull request merged**